### PR TITLE
Jg jenkins pipeline

### PR DESCRIPTION
**Pull request by mrjgreen**
Jenkinsfile for automatic builds on new CI server

### CODE REVIEW

**Comment by mrjgreen** [comment]
I'll squash this when merging into master

**Comment by jpallen** [comment]
Can you do a bit of a write up to talk me through this please? I'm not quite sure what I'm looking at and meant to review here.

File: Jenkinsfile [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+        sh 'tar -czf build.tar.gz --exclude=build.tar.gz --exclude-vcs .'
+      }
+    }
+    stage('Publish') {
+      steps {
+        withCredentials([usernamePassword(credentialsId: 'S3_CI_BUILDS_AWS_KEYS', passwordVariable: 'AWS_SECRET', usernameVariable: 'AWS_ID')]) {
+          sh '''docker run --rm \
+          -e AWS_ACCESS_KEY_ID=$AWS_ID \
+          -e AWS_SECRET_ACCESS_KEY=$AWS_SECRET \
+          -v $WORKSPACE:/app --workdir /app mrjgreen/docker-awscli s3 cp build.tar.gz s3://${S3_BUCKET_BUILD_ARTEFACTS}/${JOB_NAME}/${BUILD_NUMBER}.tar.gz
```
Comment (jpallen) [inline Jenkinsfile]: This shouldn't be namespaced as 'mrjgreen' going forward. Can we set up a `sharelatex` or `overleaf` account?

File: Jenkinsfile [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+          '''
+        }
+      }
+    }
+  }
+  
+  post {
+    failure {
+      mail(from: "team@sharelatex.com", 
+           to: "joe@sharelatex.com",
```
Comment (jpallen) [inline Jenkinsfile]: Should also be a team wide thing?

File: Jenkinsfile (hunk 1 above)
Comment (mrjgreen) [reply Jenkinsfile]: 👍 yep, will speak to you about setting up a dockerhub account for this kind of thing

**Review by jpallen — APPROVED** [review]

**Pull request merged**